zerver/lib/test_runner.py

```diff
@@ -407,6 +409,17 @@ def setup_test_environment(self, *args: Any, **kwargs: Any) -> Any:
         settings.DATABASES['default']['NAME'] = settings.BACKEND_DATABASE_TEMPLATE
         # We create/destroy the test databases in run_tests to avoid
         # duplicate work when running in parallel mode.
+
+        # Write the template ids to a file for reference during clean up
+        os.makedirs(get_dev_uuid_var_path() + '/zulip_test_template', exist_ok=True)
+        filepath = "{root}{dir}{filename}".format(
+            root=get_dev_uuid_var_path(),
+            dir='/zulip_test_template/',
+            filename=str(random.randint(1000000, 9999999))
+        )
+        with open(filepath, "w") as f:
+            for index in range(self.parallel):
+                f.write(str(random_id_range_start + (index + 1)) + "\n")
```

Comment:
We need to take care of serial mode here separately. Think about the serial case where wrong id gets written to the file.

Reply:
Pushed an update, sorry for the oversight. :)

Reply:
no problems. Reviews are exactly meant for this purpose :)